### ISSUE TITLE
Show edit links only when user has edit privilege (or is admin)

### ISSUE THREAD

**Issue by slgoldberg** [beeman/loopback-angular-admin#175]
There are many cases where relying on a 401 interceptor to present the login view is "too late" both from a usability perspective, and from a complexity perspective (i.e., it turns some state transition logic on its side).

In the [README "TODO" notes](https://github.com/beeman/loopback-angular-admin#todo) for this project, you mention two wish-list items that seem to map directly to this issue:
> Permissions on user actions (non-admins cannot access advanced functions)
> permissions on content items (non-admins can only edit own content, etc)

I am guessing this turned out to be a **lot** harder than you originally hoped, and that's why there isn't (or doesn't appear to be) anything relating to that in this code.

I have had an extremely difficult time navigating StrongLoop's documentation relating to this -- and with other similarly poorly-documented features of LoopBack, I've looked to this code base to answer many of those questions.  So it would be nice if this were also addressed here.

In short, it doesn't seem as if there's any built-in REST endpoint (e.g. `Model.canRead()` or `Model.ACL.checkPermission()` or even `ACL.checkPermission()`, just as some random examples) that can be queried from the client, in order to ask in advance whether the currently signed-in user has access to a given model, endpoint or property.

Someone who is really using LoopBack **must** have solved this problem, right?  It seems like the only other solution is to reinvent the wheel, i..e build some weird back-alley ACL-checker, for example, using some overly complicated state machine on top of the interceptor logic, to be able to probe access before setting each view). Doesn't seem tenable, but I thought I'd ask @beeman .

Thanks.

**Comment by beeman** [comment]
Hi @slgoldberg, thanks for you message

> I am guessing this turned out to be a lot harder than you originally hoped, and that's why there isn't (or doesn't appear to be) anything relating to that in this code.

The main reason that I haven't implemented this is, or any ACL at all, is because I have been too busy with my job over the last few months. I've learned a lot during the last year and I hope to be able to release an improved version of this project over the next few months. Currently it's not realistic to set any date though, I'm working on this in my free moments in the weekends.

The company I work for open sourced [this component](https://github.com/fullcube/loopback-component-access-groups) that we use in our application. This component allows us to have fine-grained access control in the app. We added a helper method to gives an overview of the users' permissions and in our Angular app we can check if a user has a certain role and hide or display things based on that output. In our Angular app we also have ACL's on our ui-router states, those use the same mechanism for checking permissions. 

Another good example of how access control can be implemented are these repo's by @pulkitsinghal 

https://github.com/ShoppinPal/multi-tenant-loopback-example
https://github.com/ShoppinPal/multi-tenant-loopback-example-two  

Let me know if you have any specific questions.

**Comment by beeman** [comment]
Closing issue as work is being done in #176
